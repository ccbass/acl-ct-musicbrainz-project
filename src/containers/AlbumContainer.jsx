import React, {useState, useEffect} from 'react';
import AlbumList from '../components/presentations/AlbumList';
import {getAlbums} from '../components/services/MusicBrainzApi';
import { useParams } from 'react-router-dom';




function AlbumContainer() {

  const [albums, setAlbum] = useState([])
  
  const [loading, setLoading] = useState(true)

  const { artist } = useParams()


  useEffect(() => { 
      getAlbums(artist)
        .then(setAlbum)
        .finally(() => 
          setLoading(false)
        )
  }, []) 

  return (
    <div>
      {loading ? (<h1>LOADING...</h1>)
      :(<AlbumList album={albums}/>)}
    </div>
  );
}

export default AlbumContainer;