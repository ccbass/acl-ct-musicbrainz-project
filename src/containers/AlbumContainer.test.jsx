import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';
import AlbumContainer from './AlbumContainer';
import AlbumData from './AlbumJson.json';

const serv = setupServer(
  rest.get('http://musicbrainz.org/ws/2/release?artist=d997d399-355e-4c49-9c7b-75a93d76bc0e&fmt=json', (req, res, ctx) => {
    return res (
      ctx.json(AlbumData));
    } 
  )
);

describe('Album Container', () => {
  beforeAll(() => serv.listen());
  afterAll(() => serv.close());
  it('renders list of albums for each artist', async () => {
    render(
    <MemoryRouter>
      <AlbumContainer/>
    </MemoryRouter>);

    const ul = await screen.findByRole('list', { name: 'album' });
    expect(ul).toMatchSnapshot();
  });
});