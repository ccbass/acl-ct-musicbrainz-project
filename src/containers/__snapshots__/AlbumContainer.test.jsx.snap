// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Container renders list of albums for each artist 1`] = `
<ul
  aria-label="album"
  data-testid="albumList"
>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/052c8908-d488-4a1f-adfc-fb621a53cc43"
      >
        <h1>
          Have Dreams!
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/052c8908-d488-4a1f-adfc-fb621a53cc43/front"
      />
      <p>
        Release Date:
        2016-04-13
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/15ac2fe9-7d68-47c6-87d5-35c865ff68f9"
      >
        <h1>
          Type 2
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/15ac2fe9-7d68-47c6-87d5-35c865ff68f9/front"
      />
      <p>
        Release Date:
        2005-07-13
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/18117c17-6a68-44b5-840d-35f949b93264"
      >
        <h1>
          Rhythm Tengoku Complete Music Collection
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/18117c17-6a68-44b5-840d-35f949b93264/front"
      />
      <p>
        Release Date:
        2008-10-15
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/2c5fa34a-9833-4d91-9bd4-b42d1885fba6"
      >
        <h1>
          V3〜青春カバー〜
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/2c5fa34a-9833-4d91-9bd4-b42d1885fba6/front"
      />
      <p>
        Release Date:
        2006-06-28
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/3f891ad1-f08c-431f-9f09-9cf0304b9790"
      >
        <h1>
          Touch me
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/3f891ad1-f08c-431f-9f09-9cf0304b9790/front"
      />
      <p>
        Release Date:
        1999-11-03
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/59e551fd-5754-458a-ba87-6e3d409dc869"
      >
        <h1>
          Love is here〜希望の光〜
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/59e551fd-5754-458a-ba87-6e3d409dc869/front"
      />
      <p>
        Release Date:
        2011-03-22
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/798691a0-e902-4977-857b-3027860e04df"
      >
        <h1>
          TAKE 1
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/798691a0-e902-4977-857b-3027860e04df/front"
      />
      <p>
        Release Date:
        2004-02-18
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/83d4aeb8-3275-4167-88a7-41849baf8f71"
      >
        <h1>
          LOVE 〜Destiny〜 / LOVE 〜since1999〜
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/83d4aeb8-3275-4167-88a7-41849baf8f71/front"
      />
      <p>
        Release Date:
        1999-04-14
      </p>
      <p>
        Quality: 
        high
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/92b741f2-e6a9-4f01-97e8-70784ffcbdac"
      >
        <h1>
          To You
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/92b741f2-e6a9-4f01-97e8-70784ffcbdac/front"
      />
      <p>
        Release Date:
        2010-12-29
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/9a9489a0-8918-4675-94ee-f10e45e19c77"
      >
        <h1>
          しょっぱいね
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/9a9489a0-8918-4675-94ee-f10e45e19c77/front"
      />
      <p>
        Release Date:
        2012-09-26
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/9f92e563-6918-3e06-bf6e-5c9409f5fb08"
      >
        <h1>
          V3〜青春カバー〜
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/9f92e563-6918-3e06-bf6e-5c9409f5fb08/front"
      />
      <p>
        Release Date:
        2006-06-28
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/a56d1452-76e5-4e2b-82a2-14d31abf3091"
      >
        <h1>
          Have Dreams!
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/a56d1452-76e5-4e2b-82a2-14d31abf3091/front"
      />
      <p>
        Release Date:
        2016-04-20
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/a6db3b6b-0ea4-40c9-812f-92749bdcf6e5"
      >
        <h1>
          A Hard Day's Night
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/a6db3b6b-0ea4-40c9-812f-92749bdcf6e5/front"
      />
      <p>
        Release Date:
        2000-12-06
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/a97eda27-cc2e-46dd-ab43-43949c2e2dc6"
      >
        <h1>
          Have Dreams!
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/a97eda27-cc2e-46dd-ab43-43949c2e2dc6/front"
      />
      <p>
        Release Date:
        2016-04-15
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/aa228eff-36b6-482f-9640-0900a6c0c4b1"
      >
        <h1>
          TAKE 1
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/aa228eff-36b6-482f-9640-0900a6c0c4b1/front"
      />
      <p>
        Release Date:
        2004-02-18
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/ad3c3bb1-b47c-4e0b-b146-04bc8a108106"
      >
        <h1>
          タイプ2
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/ad3c3bb1-b47c-4e0b-b146-04bc8a108106/front"
      />
      <p>
        Release Date:
        2005-07-13
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/b53b83d3-2f2f-4b6d-b9a1-5ec98f9934cf"
      >
        <h1>
          LOVE～抱き合って～
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/b53b83d3-2f2f-4b6d-b9a1-5ec98f9934cf/front"
      />
      <p>
        Release Date:
        1999-04-28
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
  <li>
    <div
      class="box"
    >
      <a
        activeclassname="selected"
        exact="true"
        href="/songs/undefined/undefined/cfb2dacb-db31-4ed3-a5d7-08eac927c75f"
      >
        <h1>
          リゾナント ブルー
        </h1>
      </a>
      <img
        src="http://coverartarchive.org/release/cfb2dacb-db31-4ed3-a5d7-08eac927c75f/front"
      />
      <p>
        Release Date:
        2008-05-02
      </p>
      <p>
        Quality: 
        normal
      </p>
    </div>
  </li>
</ul>
`;
